src/main/kotlin/org/rust/lang/core/psi/ext/RsTypeInferenceRoot.kt

```diff
@@ -0,0 +1,8 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.psi.ext
+
+interface RsTypeInferenceRoot : RsElement
```

Comment:
maybe someone can suggest a better name for it?

Reply:
Regardless of the name, it probably needs a doc string and a mention in the Architecture.md.

However, I find `TypeInferenceRoot` name to be pretty great actually!